File: sdks/csharp/PgChangeFeed.Client/Http/Models/Retention.cs
using System.Text.Json.Serialization;

namespace PgChangeFeed.Client.Http.Models;

/// <summary>
/// The retention run to start: deletes the changes of <c>Source</c> that are
/// older than <c>MinAgeNanos</c> and that every consumer with a stored
/// position has passed. <c>Source</c> is mandatory, <c>MinAgeNanos</c> must be
/// &gt;= 0.
/// </summary>
public sealed record RunRetentionRequest(
    [property: JsonPropertyName("source")] string Source,
    [property: JsonPropertyName("min_age_nanos")] long MinAgeNanos);

/// <summary>
/// The number of changes actually deleted.
/// </summary>
public sealed record RunRetentionResponse(
    [property: JsonPropertyName("deleted")] int Deleted);

File: sdks/kotlin/pgchangefeed-kotlin/src/main/kotlin/io/github/pt9912/pgchangefeed/http/model/Retention.kt
package io.github.pt9912.pgchangefeed.http.model

import com.google.gson.annotations.SerializedName

/**
 * The retention run to start: deletes the changes of `source` that are older
 * than `minAgeNanos` and that every consumer with a stored position has
 * passed. `source` is mandatory, `minAgeNanos` must be >= 0.
 */
data class RunRetentionRequest(
    @SerializedName("source") val source: String,
    @SerializedName("min_age_nanos") val minAgeNanos: Long,
)

/** The number of changes actually deleted. */
data class RunRetentionResponse(
    @SerializedName("deleted") val deleted: Int,
)
